### PR TITLE
Slider: add Angular, Vue and React code snippets

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: concepts/05 Widgets/Slider/00 Overview.md [hunk 1]
```diff
@@ -6,28 +6,249 @@ The **Slider** is a widget that allows an end user to set a numeric value on a c
...
+
+---
+
 In addition, you can specify the step of **Slider** values using the [step](/api-reference/10%20UI%20Widgets/dxSliderBase/1%20Configuration/step.md '/Documentation/ApiReference/UI_Widgets/dxSlider/Configuration/#step') option.
 
+---
+##### jQuery
+
+    <!--HTML-->
+    <div id="sliderContainer"></div>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/00 Overview.md]: There's no need for this since we've already shown how to declare a container for the widget.

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 2]
```diff
@@ -14,8 +17,104 @@ The **Slider** can display labels for the [min](/api-reference/10%20UI%20Widgets
         });
     });
 
+##### Angular
+
+    <!--HTML-->
+    <dx-slider
+        [label]="label"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: Use nested components whenever possible.

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 3]
```diff
@@ -41,6 +239,83 @@ To specify whether or not the part of the scale from the beginning to the slider
         });
     });
 
+##### Angular
+
+    <!--HTML-->
+    <dx-slider
+        [showRange]="showRange"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: There's no need to bind every attribute to a component property. Do not bind those that aren't usually changed at runtime.

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 4]
```diff
@@ -41,6 +239,83 @@ To specify whether or not the part of the scale from the beginning to the slider
...
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Slider } from 'devextreme-react/slider';
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                showRange: false
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: Same thing here - use state only for those properties that are going to be changed.

File: concepts/05 Widgets/Slider/10 Handle the Value Change Event.md [hunk 5]
```diff
@@ -1,28 +1,135 @@
...
+                />
+            );
+        }
+    }
+
+    export default App;
+
+---
+
 If you are going to change event handlers at runtime, or if you need to attach several handlers to the value change event, subscribe to this event using the [on(eventName, eventHandler)](/api-reference/10%20UI%20Widgets/Component/3%20Methods/on(eventName_eventHandler).md '/Documentation/ApiReference/UI_Widgets/dxSlider/Methods/#oneventName_eventHandler') method.
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/10 Handle the Value Change Event.md]:
```suggestion
If you are going to change event handlers at runtime, or if you need to attach several handlers to the value change event, subscribe to this event using the [on(eventName, eventHandler)](/api-reference/10%20UI%20Widgets/Component/3%20Methods/on(eventName_eventHandler).md '/Documentation/ApiReference/UI_Widgets/dxSlider/Methods/#oneventName_eventHandler') method. This approach is more typical of jQuery.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Slider/00 Overview.md (hunk 1 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/00 Overview.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md (hunk 2 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md (hunk 3 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md (hunk 4 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/10 Handle the Value Change Event.md (hunk 5 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/10 Handle the Value Change Event.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 6]
```diff
@@ -14,8 +17,104 @@ The **Slider** can display labels for the [min](/api-reference/10%20UI%20Widgets
...
 
+##### Angular
+
+    <!--HTML-->
+    <dx-slider
+        <dxo-label
+            [visible]="true"
+            [format]="format"
+            position="bottom"
+        ></dxo-label>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
            position="bottom"> <!-- or "top" -->
        </dxo-label>
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 7]
```diff
@@ -14,8 +17,104 @@ The **Slider** can display labels for the [min](/api-reference/10%20UI%20Widgets
...
+            DxSliderModule
+        ],
+        // ...
+    })
+
+##### Vue
+
+    <template>
+        <DxSlider
+            :label="label"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: Use nested components. Please fix this in all these examples.

File: concepts/05 Widgets/Slider/00 Overview.md [hunk 8]
```diff
@@ -6,28 +6,246 @@ The **Slider** is a widget that allows an end user to set a numeric value on a c
...
             value: 25
         });
     });
 
+##### Angular
+
+    <!--HTML-->
+    <dx-slider
+        [min]="minValue"
+        [max]="maxValue"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/00 Overview.md]: It's unlikely that users will change min, max, or step. The only Slider option that needs binding to a component or state property is value. Please update all these examples.

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md (hunk 6 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/00 Overview.md (hunk 8 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/00 Overview.md]: thx for your patience))

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md (hunk 7 above)
Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 9]
```diff
@@ -31,8 +129,106 @@ The **Slider** can also display a tooltip for the slider handle. To configure it
...
+            super(props);
+        }
+
+        render() {
+            return (
+                <Slider>
+                    <Tooltip
+                        enabled={true}
+                        position="bottom" // or "top"
+                        showMode="always" // or "onHover"
```
Comment (AlexJameson) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: I see two comments here that suggest different options. Why did you add them here but skipped in Angular and Vue?

Comment (ksercs) [reply concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]: fixed

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 10]
```diff
@@ -1,38 +1,234 @@
...
+
+    const format = function(value) {
+        return value + " units";
+    }
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+        }
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 11]
```diff
@@ -1,38 +1,234 @@
...
+        render() {
+            return (
+                <Slider>
+                    <Label
+                        visible={true}
+                        format={format}
+                        posible="bottom"> {/* or "top" */}
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
                        position="bottom" /> {/* or "top" */}
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 12]
```diff
@@ -1,38 +1,234 @@
...
+                    </Label>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion

```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 13]
```diff
@@ -1,38 +1,234 @@
...
+
+##### Vue
+
+    <template>
+        <DxSlider>
+            <DxLabel
+                :visible="true"
+                :format="format"
+                position="bottom"> <!-- or "top" -->
+            </DxLabel>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
                position="bottom" /> <!-- or "top" -->
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 14]
```diff
@@ -1,38 +1,234 @@
...
+##### Vue
+
+    <template>
+        <DxSlider>
+            <DxTooltip
+                :enabled="true"
+                position="bottom"
+                :format="format"
+                showMode="always"> <!-- or "onHover" -->
+            </DxTooltip>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
                show-mode="always" /> <!-- or "onHover" -->
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 15]
```diff
@@ -1,38 +1,234 @@
...
+
+    const format = function(value) {
+        return value + " units";
+    }
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+        }
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 16]
```diff
@@ -1,38 +1,234 @@
...
+        render() {
+            return (
+                <Slider>
+                    <Tooltip
+                        enabled={true}
+                        position="bottom"
+                        format={format}
+                        showMode="always"> {/* or "onHover" */}
+                    </Tooltip>
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
                        showMode="always" /> {/* or "onHover" */}
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 17]
```diff
@@ -41,6 +237,74 @@ To specify whether or not the part of the scale from the beginning to the slider
...
+            DxSliderModule
+        ],
+        // ...
+    })
+
+##### Vue
+
+    <template>
+        <DxSlider
+            :showRange="false"
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
            :show-range="false"
```

File: concepts/05 Widgets/Slider/05 Customize Widget Appearance.md [hunk 18]
```diff
@@ -41,6 +237,74 @@ To specify whether or not the part of the scale from the beginning to the slider
...
+    import 'devextreme/dist/css/dx.common.css';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import { Slider } from 'devextreme-react/slider';
+
+    class App extends React.Component {
+        constructor(props) {
+            super(props);
+        }
+
```
Comment (RomanTsukanov) [inline concepts/05 Widgets/Slider/05 Customize Widget Appearance.md]:
```suggestion
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Review by AlexJameson — APPROVED** [review]
